### ISSUE TITLE
Differences to FauxAPI

### ISSUE THREAD

**Issue by imphil** [jaredhendrickson13/pfsense-api#128]
Thanks for putting work into implementing an API for pfSense. I'm looking for an API and I wonder: what are the main differences between this project and https://github.com/ndejong/pfsense_fauxapi?

**Comment by jaredhendrickson13** [comment]
Hey!

Thanks for your patience. This is a great question. One of the primary reasons I created this API was I could not get approval to use FauxAPI in the enterprise setting I was working in at the time. Managers thought the design of FauxAPI was risky and too difficult to train users on. I used FauxAPI for years and it truly is a great option, however it is not always the best option for everyone. Comparatively I believe this API offers these advantages:

- Easy to use: Structured more like a traditional REST API that most developers will immediately understand. Detailed documentation is included and embedded within the API's configuration page in the pfSense webConfigurator.
- Customizable: Several authentication modes are available, and other advanced features allow administrators to fine tune how the API behaves.
- Extendable: The API is built upon a fairly simple framework that is well documented within the [contributors guide](https://github.com/jaredhendrickson13/pfsense-api/blob/master/docs/CONTRIBUTING.md). This allows users to develop their own API endpoints to serve whatever purpose they require. 
- Safe: A large problem with FauxAPI is it puts most of the input validation work on the user rather than the server. pfSense API validates input on the server-side to prevent invalid configurations from being made. For the most part, input is validated the same as it is in the pfSense webConfigurator.
- Continuous: Designed to look and behave like a native API for pfSense would be expected to. The API configuration page uses the same UI framework the rest of the pfSense webConfigurator is built on, relies on the same functions used by the webConfigurator to apply changes, etc. This makes the project easier to maintain and more reliable.

I'd suggest giving both solutions a try and see what works best for you. Feel free to reach out if you have any questions or issues.

Thanks!
